src/runner/PlutoRunner.jl

```diff
@@ -340,7 +426,14 @@ If the third argument is a `Tuple{Set{Symbol}, Set{Symbol}}` containing the refe
 
 This function is memoized: running the same expression a second time will simply call the same generated function again. This is much faster than evaluating the expression, because the function only needs to be Julia-compiled once. See https://github.com/fonsp/Pluto.jl/pull/720
 """
-function run_expression(m::Module, expr::Any, cell_id::UUID, function_wrapped_info::Union{Nothing,Tuple{Set{Symbol},Set{Symbol}}}=nothing, contains_user_defined_macrocalls::Bool=false)
+function run_expression(m::Module, expr::Any, cell_id::UUID, function_wrapped_info::Union{Nothing,Tuple{Set{Symbol},Set{Symbol}}}=nothing, forced_expr_id::Union{ObjectID,Nothing}=nothing; user_requested_run::Bool=true)
+    if user_requested_run
+        # TODO Time elapsed? Possibly relays errors in cleanup function?
+        UseEffectCleanups.trigger_cleanup(cell_id)
+
+        # TODO Could also put explicit `try_macroexpand` here, to make clear that user_requested_run => fresh macro identity
```

Comment:
![no_explicit_rerun](https://user-images.githubusercontent.com/9824244/140103979-cde8497c-9a6d-441f-aee3-772463b692a1.gif)

There seems to be a problem with this, explicitely re-running does not regenerate the underlying expression. The same goes for updating the macro definition.

Reply:
Hmm, I don't have that D:

You didn't have that before? Maybe it's time I put the explicit re-macroexpand-ization there anyway then.

Reply:
The following condition is alway false in my example:

https://github.com/fonsp/Pluto.jl/blob/5c34b99d756e7d77d624f3155b7e9007ac1b3cc4/src/runner/PlutoRunner.jl#L453

which doesn't trigger a new expansion.

Reply:
Yeah, it should, as that *should* already be handled by macro_expand from the Expression Explorer.. Wonder why it isn't for you though D:

Reply:
ah yes you're right

Reply:
okay weird, i haven't been able to reproduce